open_event/static/js/admin/event/scheduling.js

```diff
@@ -0,0 +1,512 @@
+            if (!isColliding) {
+                updateCounterBadge();
+                updateElementTime($sessionElement);
+                updateColor($sessionElement);
+            } else {
+                $sessionElement.appendTo($(".sessions-holder"));
+                $sessionElement.addClass("unscheduled").removeClass("scheduled").tooltip("hide").attr("data-original-title", "");
+                $sessionElement.css({
+                    "-webkit-transform": "",
+                    "transform": "",
+                    "background-color": ""
+                }).removeData("x").removeData("y");
+            }
+
+        },
+        ondropdeactivate: function (event) {
+            var $trackDropZone = $(event.target);
+            var $sessionElement = $(event.relatedTarget);
+            $trackDropZone.removeClass("drop-now").removeClass("drop-active");
+            if (!$sessionElement.hasClass("scheduled")) {
+                $sessionElement.css({
+                    "-webkit-transform": "",
+                    "transform": "",
+                    "background-color": ""
+                }).removeData("x").removeData("y").tooltip("hide").attr("data-original-title", "");
+            }
+        }
+    });
+}
+
+function minutesToPixels(minutes, forTop) {
+    minutes = Math.abs(minutes);
+    if (forTop) {
+        return ((minutes / time.unit.minutes) * time.unit.pixels) + time.unit.pixels;
+    } else {
+        return (minutes / time.unit.minutes) * time.unit.pixels;
+    }
+}
+
+function pixelsToMinutes(pixels, fromTop) {
+    pixels = Math.abs(pixels);
+    if (fromTop) {
+        return ((pixels - time.unit.pixels) / time.unit.pixels) * time.unit.minutes;
+    } else {
+        return (pixels / time.unit.pixels) * time.unit.minutes;
+    }
+}
+
+var days = [];
+var sessionsStore = [];
+var tracksStore = [];
+
+function processTrackSession(tracks, sessions, callback) {
+
+    var topTime = moment.utc({hour: time.start.hours, minute: time.start.minutes});
+
+    _.each(sessions, function (session) {
+
+        var startTime = moment.utc(session.start_time);
+        var endTime = moment.utc(session.end_time);
+        var duration = moment.duration(endTime.diff(startTime));
+
+        var top = minutesToPixels(moment.duration(moment.utc({
+            hour: startTime.hours(),
+            minute: startTime.minutes()
+        }).diff(topTime)).asMinutes(), true);
+
+        console.log(session.start_time);
+
+        var dayString = startTime.format("Do MMMM YYYY"); // formatted as eg. 2nd May
+
+        if (!_.contains(days, dayString)) {
+            days.push(dayString);
+        }
+
+        var sessionObject = {
+            id: session.id,
+            title: session.title,
+            top: top,
+            duration: Math.abs(duration.asMinutes()),
+            track_id: session.track.id,
+            start_time: startTime.format("HH:mm"),
+            end_time: endTime.format("HH:mm")
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'end_time' is not in camel case.](https://www.codacy.com/app/triplez/open-event-orga-server/file/2650806054/issues/source?bid=3233864&fileBranchId=3324174#l341)